python/common/org/python/types/Float.java

```diff
@@ -352,39 +365,16 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
                 if (other_val == 0) {
                     throw new org.python.exceptions.ZeroDivisionError("integer division or modulo by zero");
                 } else {
-                    double result = this.value % other_val;
-                    if (other_val > 0 && result < 0) {
-                        // second operand is positive, ensure that result is positive
-                        result += other_val;
-                    } else if (other_val < 0 && result > 0) {
-                        // second operand is negative, ensure that result is negative
-                        result += other_val; // subtract other_val, which is negative
-                    }
-                    if (other_val < 0 && result >= 0
-                            && Double.doubleToRawLongBits(result) != NEGATIVE_ZERO_RAW_BITS) {
-                        result *= -1;
-                    }
+                    // Reference: http://stackoverflow.com/a/4412200
+                    double result = (((((double) this.value) % other_val) + other_val) % other_val);
                     return new org.python.types.Float(result);
                 }
             } else if (other instanceof org.python.types.Float) {
                 double other_val = ((org.python.types.Float) other).value;
                 if (other_val == 0.0) {
                     throw new org.python.exceptions.ZeroDivisionError("float modulo");
                 } else {
-                    double result = ((double) this.value) % other_val;
-                    if (other_val > 0.0 && result < 0.0) {
-                        // second operand is positive, ensure that result is positive
-                        result += other_val;
-                    } else if (other_val < 0.0 && result > 0.0) {
-                        // second operand is negative, ensure that result is negative
-                        result += other_val; // subtract other_val, which is negative
-                    }
-                    // edge case where adding -0.0 to 0.0 doesn't yield the expected -0.0
-                    // do this only if it is definitely not -0.0 already
-                    if (other_val < 0.0 && result >= 0.0
-                            && Double.doubleToRawLongBits(result) != NEGATIVE_ZERO_RAW_BITS) {
-                        result *= -1;
-                    }
+                    double result = (((((double) this.value) % other_val) + other_val) % other_val);
```

Comment:
Yes, it does work for float. The test that failed was a different one, I'm fixing that right now.
Though the same method does throw an error for Int:
>>> -3.14159%-5
-3.14159
However, my code returns
-3.1415900000000008
I'm trying to fix it, but I'm not really sure how to proceed from here though.